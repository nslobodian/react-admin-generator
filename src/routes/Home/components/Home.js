import React from 'react'
import s from './Home.scss'

export const Home = () => (
  <div className={s.Home}>
    <div className={s.blackSquare}><div className={s.text}>Admin Panel</div></div>
  </div>
)

export default Home
