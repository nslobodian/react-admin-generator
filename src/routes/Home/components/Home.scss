.Home {
  .blackSquare {
    margin: 150px auto;
    width: 400px;
    height: 400px;
    background-color: #fff;
    color: #fff;
    font-size: 40px;
    text-align: center;
    line-height: 400px;

    .text {
      &:hover {
        transition-property: color, background-color;
        transition-duration: 5s;
        color: #fff;
        background-color: #000;
      }
    }
  }
}

