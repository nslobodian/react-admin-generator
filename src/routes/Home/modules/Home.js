import createReducer from 'utils/createReducer'

// ------------------------------------
// Constants
// ------------------------------------

export const HOME_FETCH = 'HOME_FETCH'

// ------------------------------------
// Actions
// ------------------------------------

export const homeFetch = (params) => (dispatch, getStore) => {
  dispatch({
    type: HOME_FETCH,
    params,
  })
}

// ------------------------------------
// Reducer
// ------------------------------------
const initialState = {
  fetching: false,
}

export default createReducer(initialState, {
  [HOME_FETCH]: (state, action) => ({
    fetching: false,
  }),
})
