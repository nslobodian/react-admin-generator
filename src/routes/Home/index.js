import { injectReducer } from 'store/reducers'

export default (store) => ({
  getComponent (nextState, cb) {
    Promise.all([
      import('./containers/HomeContainer'),
      import('./modules/Home'),
    ]).then((modules) => {
      const Home = modules[0].default
      const reducer = modules[1].default

      injectReducer(store, { key: 'Home', reducer })

      cb(null, Home)
    })
  },
})
