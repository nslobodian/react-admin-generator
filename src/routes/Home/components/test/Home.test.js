import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Home } from '../Home'

describe('(Routes) Home', () => {
  it('Should render Home properly', () => {
  const props = {}
    const wrapper = shallow(
      <Home
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
