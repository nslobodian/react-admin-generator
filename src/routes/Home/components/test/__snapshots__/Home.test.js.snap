// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Routes) Home Should render Home properly 1`] = `
<div
  className="Home"
>
  <h4>
    Home
  </h4>
</div>
`;
